### PR TITLE
fix: show the language in the three dots menu

### PR DESCRIPTION
**Pull request by edmonday**
# Description

### Issue

Show the language in the three dots menu.

### Solution

- Added the journey language to `JourneyDetails.tsx`
- Refactored code in `Toolbar.tsx` to use `JourneyDetails.tsx`
- Moved `JourneyDetails` and `JourneyDetailsDialog` folder

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  it('should display title, language and description', () => {
+    render(
+      <JourneyProvider value={{ journey: mockJourney }}>
+        <JourneyDetails />
+      </JourneyProvider>
+    )
+    expect(screen.getByText('Some title')).toBeInTheDocument()
+    expect(screen.getByText('English')).toBeInTheDocument()
+    expect(screen.queryByTestId('DescriptionDot')).toBeInTheDocument()
+    expect(screen.getByText('Some description')).toBeInTheDocument()
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.spec.tsx:40]: Can we please use `getByRole('heading',....)` instead? Jest suggests an order of priority in terms of which query to use when testing

File: apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  })
+
+  it('should not show dot if there is no description', () => {
+    render(
+      <JourneyProvider value={{ journey: noDescriptionJourney }}>
+        <JourneyDetails />
+      </JourneyProvider>
+    )
+    expect(screen.queryByTestId('DescriptionDot')).not.toBeInTheDocument()
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.spec.tsx:49]: should also check for description to not be there?

File: apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+export const Default = {
+  ...Template,
+  parameters: {
+    apolloClient: {
+      mocks: [getLanguagesMock]
+    }
+  }
+}
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.stories.tsx:98]: Let me know if I'm not understanding the styling thing. But was thinking default should have short text, then ellipsis is for long texts (title, description)

File: apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+import { ReactElement } from 'react'
+
+import { useJourney } from '@core/journeys/ui/JourneyProvider'
+import Globe1Icon from '@core/shared/ui/icons/Globe1'
+
+export function JourneyDetails(): ReactElement {
+  const { journey } = useJourney()
+  const language = journey?.language.name.find(
+    ({ primary }) => primary != null
+  )?.value
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.tsx:14]: Looks good though we should check if primary is true or false instead. We have a pattern set for it if you search up localName, e.g. localName ?? nativeName

File: apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+  return (
+    <>
+      {journey != null ? (
+        <Stack
+          data-testid="JourneyDetails"
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.tsx:20]: do we need this testid?

File: apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        >
+          <Typography
+            sx={{
+              display: { xs: '-webkit-box', md: 'unset' },
+              '-webkit-line-clamp': { xs: '2', md: '1' },
+              '-webkit-box-orient': 'vertical',
+              overflow: 'hidden',
+              textOverflow: 'ellipsis',
+              color: 'secondary.dark',
+              typography: { xs: 'subtitle2', md: 'subtitle1' }
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.tsx:36]: this is cool!

File: apps/journeys-admin/src/components/Editor/Toolbar/Menu/Menu.spec.tsx [hunk 7]
```diff
@@ -302,10 +302,7 @@ describe('Toolbar Menu', () => {
         </SnackbarProvider>
       )
       fireEvent.click(screen.getByRole('button'))
-      expect(
-        screen.getByRole('heading', { level: 6, name: 'some title' })
-      ).toBeInTheDocument()
-      expect(screen.getByText('some description')).toBeInTheDocument()
+      expect(screen.getByTestId('JourneyDetails')).toBeInTheDocument()
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Toolbar/Menu/Menu.spec.tsx:305]: we can keep the test to be the same here

File: apps/journeys-admin/src/components/Editor/Toolbar/Menu/Menu.tsx [hunk 8]
```diff
@@ -41,7 +39,7 @@ interface MenuProps {
 }
 
 export function Menu({ user }: MenuProps): ReactElement {
-  const smUp = useMediaQuery((theme: Theme) => theme.breakpoints.up('sm'))
+  const mdUp = useMediaQuery((theme: Theme) => theme.breakpoints.up('md'))
   const { journey } = useJourney()
   const { t } = useTranslation('apps-journeys-admin')
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Toolbar/Menu/Menu.tsx:44]: NIT: hahaha this is more of a John thing, but can you please move the `t` function as the first variable defined

File: apps/journeys-admin/src/components/Editor/Toolbar/Toolbar.spec.tsx [hunk 9]
```diff
@@ -147,20 +139,9 @@ describe('Toolbar', () => {
...
-    expect(
-      screen.getAllByText('My Awesome Journey Description')[0]
-    ).toBeInTheDocument()
-    expect(screen.getAllByRole('paragraph')[0]).toHaveTextContent('English')
-  })
-
-  it('should not show dot if there is no description', () => {
-    render(toolbar(noDescriptionJourney))
-    expect(screen.queryByTestId('DescriptionDot')).not.toBeInTheDocument()
+    expect(screen.getByTestId('JourneyDetails')).toBeInTheDocument()
```
Comment (GeronimoJohn) [inline apps/journeys-admin/src/components/Editor/Toolbar/Toolbar.spec.tsx:144]: Can keep test to be the same. TestId is kinda just checking if the div renders

File: apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.spec.tsx (hunk 1 above)
Comment (edmonday) [reply apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.spec.tsx:40]: they are not headings. they are all p tags

Comment (GeronimoJohn) [reply apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.spec.tsx:40]: https://testing-library.com/docs/queries/about/#priority

Comment (GeronimoJohn) [reply apps/journeys-admin/src/components/Editor/Toolbar/JourneyDetails/JourneyDetails.spec.tsx:40]: Oh yes you've mentioned this

**Review by GeronimoJohn — APPROVED** [review]

**Review by GeronimoJohn — APPROVED** [review]

**Review by GeronimoJohn — APPROVED** [review]

**Pull request merged**